slick/src/main/scala/akka/stream/alpakka/slick/scaladsl/Slick.scala

```diff
@@ -76,6 +76,44 @@ object Slick {
         session.db.run(toStatement(t))
       }
 
+  /**
+   * Scala API: creates a Flow that takes a stream of elements of
+   *            type T, transforms each element to a SQL statement
+   *            using the specified function, then executes
+   *            those statements against the specified Slick database
+   *            and returns the statement result type R.
+   *
+   * @param toStatement A function to produce the SQL statement to
+   *                    execute based on the current element.
+   * @param session The database session to use.
+   */
+  def flowWithPassThrough[T, R](
+      toStatement: T => DBIO[R]
+  )(implicit session: SlickSession): Flow[T, R, NotUsed] = flowWithPassThrough(1, toStatement)
+
+  /**
+   * Scala API: creates a Flow that takes a stream of elements of
+   *            type T, transforms each element to a SQL statement
+   *            using the specified function, then executes
+   *            those statements against the specified Slick database
+   *            and returns the statement result type R.
+   *
+   * @param toStatement A function to produce the SQL statement to
+   *                    execute based on the current element.
+   * @param parallelism How many parallel asynchronous streams should be
+   *                    used to send statements to the database. Use a
+   *                    value of 1 for sequential execution.
+   * @param session The database session to use.
+   */
+  def flowWithPassThrough[T, R](
+      parallelism: Int,
+      toStatement: T => DBIO[R]
+  )(implicit session: SlickSession): Flow[T, R, NotUsed] =
+    Flow[T]
+      .mapAsync(parallelism) { t =>
+        session.db.run(toStatement(t))
+      }
```

Comment:
Isn't this identical to `flow` now, so this is not needed. 
Alternatively we provide a signature just like the Java one which takes a mapper.

Also makes me wonder, do we really need the Java one, can't that also be done by closing over state in the `toStatement` function?

Reply:
`flowWithPassThrough` is indeed the same implementation as `flow`, but yes it is needed since `flow` is fixed on the return type and `flowWithPassThrough` isn't. But since implementation is the same, changed `flow` implementation to call `flowWithPassThrough` function.

The Java method is also needed, IMHO, since the 'problem' with the Java implementation is that the `toStatement` function is a function from the input `T` into a Slick `String` to not expose the Slick Scala `DBIO` type in the Java API I suppose. So the `toStatement` cannot be a function from `T` to `R` and the additional `mapper` function is needed there.

Reply:
Ah, I see, thanks for the explanation.